app/models/normalize_primo_articles.rb

```diff
@@ -61,4 +55,21 @@ def openurl
       @record['delivery']['almaOpenurl']
     end
   end
+
+  # Here we are converting the Alma link resolver URL provided by the Primo 
+  # Search API to redirect to the Primo UI. This is done for UX purposes, 
+  # as the regular Alma link resolver URLs redirect to a plaintext 
+  # disambiguation page.
+  def construct_primo_openurl
+    return unless @record['delivery']['almaOpenurl']
+
+    primo_openurl_base = [ENV['MIT_PRIMO_URL'], '/discovery/openurl?institution=', ENV['EXL_INST_ID'], '&vid=',
+                          ENV['PRIMO_VID'], '&'].join('')
+    primo_openurl = @record['delivery']['almaOpenurl'].gsub(ENV['ALMA_OPENURL'], primo_openurl_base)
+
+    # The ctx params appear to break Primo openurls, so we need to remove them.
+    params = Rack::Utils.parse_nested_query(primo_openurl)
+    filtered = params.delete_if { |key, value| key.include?('ctx') }
```

Comment:
It feels like there is an edge case that I just want to make sure we are comfortable with. I think we are, but want to note it. A parameter of `rft.ctxblahblahblah` would be scrubbed. I think that is unlikely enough to not care about, but wanted to be sure you agree this implementation is good enough and a `start_with?` instead of `include?` would not be better based on what we know.

Reply:
That's a good point. I think `start_with?` is a good idea for now.

Reply:
Hm...actually, I'm having second thoughts about this. I think we would want to scrub, for example, something like this: `url_ctx_fmt=info:ofi/fmt:kev:mtx:ctx`

Reply:
Yeah not knowing what Primo can actually handle is annoying. I'm comfortable with however we handle this as regardless of what we do we'll likely